File: Algorithm/Python/1000+/1833. Maximum Ice Cream Bars.py
class Solution(object):
    def maxIceCream(self, costs, coins):
        """
        :type costs: List[int]
        :type coins: int
        :rtype: int
        """
        costs.sort()
        result = 0
        for cost in costs:
            if cost > coins:
                break
            result += 1
            coins -= cost
        return result
                

File: Algorithm/C++/1000+/1833. Maximum Ice Cream Bars.cc
#include <algorithm>
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        std::sort(costs.begin(), costs.end());
        int result = 0;
        for(int c : costs) {
            if (coins < c) {
                break;
            }
            coins -= c;
            result++;
        }
        return result;
    }
};